### DISCUSSION TITLE
kdb and db files

### DISCUSSION THREAD

**Discussion by waheedi** [ROCmSoftwarePlatform/MIOpen#3163] [Q&A] [answered]
Hello folks,

I have been trying to create a stack locally where it can support the gfx1010, I have been trying to understand where does the gfx1030.kdb and the like comes from, I know the db files are generated using sqlite which have two tables config and perf_table, I also know the kdb files are in this repository as files, but how were they generated?

All my best,

**Comment by waheedi** [comment]
As far as I can read and understand, they are meant to improve runtime loading and quicker reference for tunings (probably more things), but missing them won't break functionality if the stack including MIOpen were built targeting a specific GPU like gfx1010 for example, the stuff would happen in runtime, but would a bit slower

**Comment by huanrwan-amd** [comment, marked as answer]
Hi @waheedi , thank you for posting and sharing your findings. 
Yes, the kernel are built in runtime, saved to dbs and queried from dbs if already built. It is for improving runtime performance. You can find more information on https://rocm.docs.amd.com/projects/MIOpen/en/latest/conceptual/cache.html.
